docs/create_allocator_service.md

```diff
@@ -0,0 +1,435 @@
+
+Now https://allocator-service.yourdomain.ext should return the IP address and port of the allocated Game Server.
+
+
+### 14. Customize
+Edit main.go to reflect your kubernetes configuration by changing which namespace is used and which fleet GameServers are being allocated from.  Also create a unique key or password that will be used when calling this allocator service.
+
+
+Change some constants to reflect your environment.  These are the current values:
+```
+const namespace    = "default"
+const fleetname    = "simple-udp"
+const generatename = "simple-udp-"
+```
+
+Add a unique password to the value for the v1GameClientKey in main.go, for example:
+```
+if key != "v1GameClientKey" || value != "7948670a-92b1-45b3-8bce-587322c07921" {
+    http.Error(w, "authorization failed", http.StatusUnauthorized)
+    return
+}
+```
+
+Since the Docker image is using Alpine Linux, the "go build" command has to include few more environment variables.
+
+Let's get our dependencies and build with these commands:
+```
+go get agones.dev/agones/pkg/client/clientset/versioned
+go get agones.dev/agones/pkg/apis/stable/v1alpha1
+go get agones.dev/agones/pkg/util/runtime
+go get k8s.io/api/core/v1
+go get k8s.io/apimachinery/pkg/apis/meta/v1
+go get k8s.io/client-go/rest
+GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o bin/server -a -v main.go
+```
+
+Build a new docker image, where REPO is your repository and TAG is your tag:
+```
+docker build -t [REPO]/allocator-service:[TAG] .
+```
+
+Push it to whatever repo you use:
+```
+docker push [REPO]/allocator-service:[TAG]
+```
+
+Edit allocator-service.yaml to point to the new image:
+```
+containers:
+- name: fleet-allocator
+  image: your.repo/user/allocator-service:tag
+  imagePullPolicy: Always
+```
+
+Edit your game client, matchmaker, or other dependent service to hit the subdomain.domain.ext/address endpoint to retrieve the IP address and port of the newly allocated Game Server.
+
+Consider adding some code to the dedicated game server that checks how many connected users there are, or how many players are in the world.  If it returns 0, call the Agones SDK /shutdown endpoint to destroy the stale Pod.
+
+
+### 15. Cleanup
+You can delete the allocator service and associated resources with the following commands.
+
+Delete the Ingress
+```
+kubectl delete ingress fleet-allocator-ingress
+```
+
+Delete the Reserved IP address
+```
+gcloud compute addresses delete allocator-static-ip --global
+```
+
+Delete the Service
+```
+kubectl delete -f allocator-service.yaml
+```
+
+Delete the Service Account
+```
+kubectl delete -f service-account.yaml
+```
+
+
+### 16. Related Documentation
+
+The [Gin Web Framework](https://github.com/gin-gonic/gin/)
+
+The Kubernetes documentation on [Services](https://kubernetes.io/docs/concepts/services-networking/service/)
```

Comment:
We could link this above, as we introduce each of these concepts.

Reply:
Sounds good.  I'll add inline links while reducing explanatory text.

Thanks for the review!